File: core/src/main/java/com/gmail/marcosav2010/peer/PeerProperties.java
package com.gmail.marcosav2010.peer;

import com.gmail.marcosav2010.config.IConfiguration;
import com.gmail.marcosav2010.config.Properties;
import com.gmail.marcosav2010.config.PropertyCategory;
import com.gmail.marcosav2010.handshake.HandshakeAuthenticator.HandshakeRequirementLevel;

public class PeerProperties extends Properties {

    public static final String HANDSHAKE_REQUIREMENT_LEVEL = "defHandshakeRequirementLevel";

    public PeerProperties(IConfiguration configuration) {
        super(PropertyCategory.PEER, configuration);
    }

    public HandshakeRequirementLevel getHRL() {
        return super.get(HANDSHAKE_REQUIREMENT_LEVEL);
    }

    public void setHRL(HandshakeRequirementLevel level) {
        super.set(HANDSHAKE_REQUIREMENT_LEVEL, level);
    }
}

File: core/src/main/java/com/gmail/marcosav2010/peer/PeerProperties.kt
package com.gmail.marcosav2010.peer

import com.gmail.marcosav2010.config.IConfiguration
import com.gmail.marcosav2010.config.Properties
import com.gmail.marcosav2010.config.PropertyCategory
import com.gmail.marcosav2010.handshake.HandshakeAuthenticator

class PeerProperties(configuration: IConfiguration) : Properties(PropertyCategory.PEER, configuration) {

    var hrl: HandshakeAuthenticator.HandshakeRequirementLevel
        get() = super.get(HANDSHAKE_REQUIREMENT_LEVEL)
        set(level) {
            super.set(HANDSHAKE_REQUIREMENT_LEVEL, level)
        }

    companion object {
        const val HANDSHAKE_REQUIREMENT_LEVEL = "defHandshakeRequirementLevel"
    }
}